client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,366 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the RegisteredPipeline class
+"""
+
+from unittest.mock import patch
+
+import pytest
+from hypothesis import given, HealthCheck, settings, strategies as st
+
+import verta
+from tests.unit_tests.strategies import resources
+from verta.pipeline import RegisteredPipeline
+
+
+def test_copy_graph(
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that the graph of a RegisteredPipeline can be copied.
+
+    Each step in the copied graph should be a new object, but have the same
+    name, predecessors, and model version as the original.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rm")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+    copied_graph = pipeline.copy_graph()
+    # convert from sets to lists and sort for side-by-side comparison
+    graph_steps_sorted = sorted(list(graph.steps), key=lambda x: x.name)
+    copied_graph_steps_sorted = sorted(list(copied_graph.steps), key=lambda x: x.name)
```

Comment:
fyi, don't need to cast to `list`

```suggestion
    graph_steps_sorted = sorted(graph.steps, key=lambda x: x.name)
    copied_graph_steps_sorted = sorted(copied_graph.steps, key=lambda x: x.name)
```